test/index.spec.ts

```diff
@@ -107,36 +104,67 @@ describe('CreateUser Mutation Test', () => {
   });
 });
 
-//Temporary test for login, it compares to a specific return, it will be changed in the next PR's.
-describe('Login Mutation test - Return', () => {
-  it('Should return be equal input', async () => {
-    const query = `mutation($email: String!, $password: String!) {
-      login(email: $email, password: $password) {
-        user {
-          id
-          name
-          email
-          birthdate
-        },
-        token
-      }
-    }`;
-
-    const email = 'mochauser@email.com';
-    const password = 'mochauserPassword1';
-
-    const result = await connection.post('/graphql', {
-      query,
-      variables: { email, password },
+describe('Login Mutation Test', () => {
+  it('Should return be the user that has the same email from input and a token', async () => {
+    const input = {
+      name: 'MochaUser1',
+      email: 'mochauser1@email.com',
+      password: 'mochauserPassword1',
+      birthdate: '01-01-2000',
+    };
+
+    await mutationCreateUser(connection, input);
+    const userInDatabase = await AppDataSource.getRepository(User).findOneBy({ email: input.email });
+
+    const loginResult = (await mutationLogin(connection, input.email, input.password)).data.login;
+
+    expect(loginResult.user).to.be.deep.eq({
+      id: userInDatabase.id,
+      name: input.name,
+      email: input.email,
+      birthdate: input.birthdate,
     });
 
-    const loginResponse = {
-      id: 12,
-      name: 'Rodrigo',
-      email: 'rodrigo@email.com',
-      birthdate: '01-01-1980',
+    expect(loginResult.token).to.be.a('string');
+    expect(loginResult.token.length).to.be.gt(0);
+
+    after(async () => {
+      await AppDataSource.getRepository(User).delete({ email: input.email });
+    });
```

Comment:
Acho que isso poderia ser simplificado. Um `afterEach` que remove todos os usuário do banco depois de cada teste, por exemplo. O que acha? Assim vc garante que o banco sempre estará limpo no início dos testes.

Reply:
Então, nem todos os testes precisam de um usuário, o afterEach seria executado em todos certo? 

Porém poderia fazer um before, adicionar um usuário teste, e então um after para deletar esse usuário, mas então ficaria a pergunta: 
-Isso não vai contra o princípio de "Cada teste deve ser independente/isolado"? 

Porque caso não vá contra, eu poderia fazer isso, assim os testes que não precisarem de um usuário teste não serão afetados.

Reply:
Os testes que não precisarem de um usuário criado, ficaria redundante mesmo?

Reply:
Não vejo tanto problema, mas ainda asism eu costumo prefirir criar dentro do teste caso hajam testes que não precisem.

Mas quanto do `afterEach`, se ele por acaso executar delete mesmo não tendo usuário, não vejo isso como um problema. Nesse caso vc estará trocando um reaproveitamento de código por alguns possíveis deletes inúteis, mas esses deletes inúteis não prejudicam em nada.

Reply:
Feito, coloquei um afterEach para deletar o usuário "mocha", mesmo se um teste não precisar ele vai deletar "nada".